core/chains/evm/txmgr/orm_test.go

```diff
@@ -608,3 +608,355 @@ func TestORM_GetInProgressEthTxAttempts(t *testing.T) {
 	assert.Len(t, attempts, 1)
 	assert.Equal(t, etx.EthTxAttempts[0].ID, attempts[0].ID)
 }
+
+func TestORM_FindEthReceiptsPendingConfirmation(t *testing.T) {
+	t.Parallel()
+
+	db := pgtest.NewSqlxDB(t)
+	cfg := newTestChainScopedConfig(t)
+	borm := cltest.NewTxmORM(t, db, cfg)
+	ethKeyStore := cltest.NewKeyStore(t, db, cfg).Eth()
+	ethClient := evmtest.NewEthClientMockWithDefaultChain(t)
+	_, fromAddress := cltest.MustInsertRandomKeyReturningState(t, ethKeyStore, 0)
+
+	pgtest.MustExec(t, db, `SET CONSTRAINTS pipeline_runs_pipeline_spec_id_fkey DEFERRED`)
+
+	head := evmtypes.Head{
+		Hash:   utils.NewHash(),
+		Number: 10,
+		Parent: &evmtypes.Head{
+			Hash:   utils.NewHash(),
+			Number: 9,
+			Parent: &evmtypes.Head{
+				Number: 8,
+				Hash:   utils.NewHash(),
+				Parent: nil,
+			},
+		},
+	}
+
+	minConfirmations := int64(2)
+
+	run := cltest.MustInsertPipelineRun(t, db)
+	tr := cltest.MustInsertUnfinishedPipelineTaskRun(t, db, run.ID)
+	pgtest.MustExec(t, db, `UPDATE pipeline_runs SET state = 'suspended' WHERE id = $1`, run.ID)
+
+	etx := cltest.MustInsertConfirmedEthTxWithLegacyAttempt(t, borm, 3, 1, fromAddress)
+	pgtest.MustExec(t, db, `UPDATE eth_txes SET meta='{"FailOnRevert": true}'`)
+	attempt := etx.EthTxAttempts[0]
+	cltest.MustInsertEthReceipt(t, borm, head.Number-minConfirmations, head.Hash, attempt.Hash)
+
+	pgtest.MustExec(t, db, `UPDATE eth_txes SET pipeline_task_run_id = $1, min_confirmations = $2 WHERE id = $3`, &tr.ID, minConfirmations, etx.ID)
+
+	receiptsPlus, err := borm.FindEthReceiptsPendingConfirmation(testutils.Context(t), head.Number, *ethClient.ChainID())
+	require.NoError(t, err)
+	assert.Len(t, receiptsPlus, 1)
+	assert.Equal(t, tr.ID, receiptsPlus[0].ID)
+}
+
+func TestORM_FindEthTxWithNonce(t *testing.T) {
+	t.Parallel()
+
+	db := pgtest.NewSqlxDB(t)
+	cfg := newTestChainScopedConfig(t)
+	borm := cltest.NewTxmORM(t, db, cfg)
+	ethKeyStore := cltest.NewKeyStore(t, db, cfg).Eth()
+	_, fromAddress := cltest.MustInsertRandomKeyReturningState(t, ethKeyStore, 0)
+
+	t.Run("returns nil if no results", func(t *testing.T) {
+		etx, err := borm.FindEthTxWithNonce(fromAddress, 777)
+		require.NoError(t, err)
+		assert.Nil(t, etx)
+	})
+
+	t.Run("returns transaction if it exists", func(t *testing.T) {
+		etx := cltest.MustInsertConfirmedEthTxWithLegacyAttempt(t, borm, 777, 1, fromAddress)
+		require.Equal(t, int64(777), *etx.Nonce)
+
+		res, err := borm.FindEthTxWithNonce(fromAddress, 777)
+		require.NoError(t, err)
+		assert.Equal(t, etx.Nonce, res.Nonce)
+	})
+}
+
+func TestORM_updateEthTxAttemptUnbroadcast(t *testing.T) {
```

Comment:
Instead of testing this function, you can just test the UpdateEthTxForRebroadcast function.
That test should validate all the database entries expecting changes.
Let us not test private methods.